File: C1.UWP.ExpressionEditor/VB/ExpressionEditorSamples/ExpressionGroupDescription.vb
Imports System.Globalization
Imports C1.Xaml.ExpressionEditor
Imports C1.Xaml

Public Class ExpressionGroupDescription
    Inherits GroupDescription
    Public Shared editer As New C1ExpressionEditor()

    Public Property Expression() As String

    Public Overrides Function GroupNameFromItem(item As Object, level As Integer, culture As CultureInfo) As Object
        editer.DataSource = item
        editer.Expression = Expression

        If editer.IsValid Then
            Return editer.Evaluate()
        End If
        Return ""
    End Function
End Class

File: C1.UWP.ExpressionEditor/CS/ExpressionEditorSamples/ExpressionGroupDescription.cs
using C1.Xaml;
using C1.Xaml.ExpressionEditor;
using System.Globalization;

namespace ExpressionEditorSamples
{
    public class ExpressionGroupDescription : GroupDescription
    {
        public static C1ExpressionEditor editer = new C1ExpressionEditor();

        public string Expression
        {
            get;
            set;
        }

        public override object GroupNameFromItem(object item, int level, CultureInfo culture)
        {
            editer.DataSource = item;
            editer.Expression = Expression;

            if (editer.IsValid)
                return editer.Evaluate();
            return "";
        }
    }
}
